libevm/VM.cpp

```diff
@@ -573,6 +573,48 @@ void VM::interpretCases()
 		}
 		NEXT
 
+#ifdef EVM_USE_BITSHIFT
+		CASE(SHL)
+		{
+			ON_OP();
+			updateIOGas();
+
+			/// This workarounds a bug in Boost, ...
+			u256 mask = (u256(1) << (256 - m_SP[1])) - 1;
```

Comment:
I don't know what happens when `m_SP[1]` is more than 256.
* Is the underflown subtraction defined?
* Isn't the shift very slow when `256 - m_SP[1]` is a big number?